### PR TITLE
Split button A11y fixes

### PR DESCRIPTION
**Pull request by dmceachernmsft**
# What

- add strings for button types -> `split button` OR `button`
  - this Follows Fluents example on their docs here: https://developer.microsoft.com/en-us/fluentui#/controls/web/button
- add default keyboard behavior by assigning a Keycode to the `menuTriggerKeyCode` prop

# Why

Allows for better narration around the split buttons, as well as better keyboard navigation.

https://skype.visualstudio.com/SPOOL/_workitems/edit/2782073
# How Tested

Ran locally and tested with windows narrator. tested keyboard navigation behavior.

### CODE REVIEW

File: packages/react-components/src/components/ControlBarButton.tsx [hunk 1]
```diff
@@ -137,6 +144,8 @@ export const ControlBarButton = (props: ControlBarButtonProps): JSX.Element => {
         onRenderIcon={props.onRenderIcon ?? DefaultRenderIcon}
         ariaLabel={props.ariaLabel ?? tooltipContent ?? labelText}
         allowDisabledFocus={props.allowDisabledFocus ?? true}
+        aria-roledescription={'split button'}
```
Comment (JamesBurnside) [inline packages/react-components/src/components/ControlBarButton.tsx:147]: should this be being set here? (was this left in?)

File: packages/react-components/src/components/CameraButton.tsx [hunk 2]
```diff
@@ -170,6 +180,11 @@ export const CameraButton = (props: CameraButtonProps): JSX.Element => {
       menuProps={props.menuProps ?? generateDefaultDeviceMenuPropsTrampoline(props, strings)}
       menuIconProps={props.menuIconProps ?? !enableDeviceSelectionMenuTrampoline(props) ? { hidden: true } : undefined}
       split={props.split ?? enableDeviceSelectionMenuTrampoline(props)}
+      ariaDescription={
```
Comment (JamesBurnside) [inline packages/react-components/src/components/CameraButton.tsx:183]: Can you explain more in the PR description what this is doing and why is it needed? Does fluent not provide the aria roles for their buttons already?

Comment (JamesBurnside) [reply packages/react-components/src/components/CameraButton.tsx:183]: Wait -- is there a difference between aria-roledescriptiona and aria-description?

File: packages/react-components/src/components/CameraButton.tsx [hunk 3]
```diff
@@ -48,6 +48,16 @@ export interface CameraButtonStrings {
    * Tooltip of camera menu
    */
   cameraMenuTooltip: string;
+  /* @conditional-compile-remove(call-with-chat-composite) @conditional-compile-remove(control-bar-split-buttons) */
```
Comment (prprabhu-ms) [inline packages/react-components/src/components/CameraButton.tsx:51]: These should only reference `control-bar-split-buttons` feature as the split buttons in `react-components` can be stabilized independently.

And any case, I think both of these features are now stabilized, so don't add conditional compilation at all: https://github.com/Azure/communication-ui-library/blob/main/common/config/babel/.babelrc.js#L30

**Comment by prprabhu-ms** [comment]
I'm confused how this fixes keyboard navigation. Comments in the code explaining what affects keyboard navigation would be helpful.

File: packages/react-components/src/components/ControlBarButton.tsx (hunk 1 above)
Comment (dmceachernmsft) [reply packages/react-components/src/components/ControlBarButton.tsx:147]: oops will fix it!

File: packages/react-components/src/components/CameraButton.tsx (hunk 2 above)
Comment (dmceachernmsft) [reply packages/react-components/src/components/CameraButton.tsx:183]: they dont provide the explicit narration of 'split button' which I think would provide more context to say the drop down is attached to this button, will update for more clarity 😊

File: packages/react-components/src/components/CameraButton.tsx (hunk 3 above)
Comment (dmceachernmsft) [reply packages/react-components/src/components/CameraButton.tsx:51]: okay will remove these comments! 😊

File: packages/react-components/src/components/ControlBarButton.tsx [hunk 4]
```diff
@@ -137,6 +144,8 @@ export const ControlBarButton = (props: ControlBarButtonProps): JSX.Element => {
         onRenderIcon={props.onRenderIcon ?? DefaultRenderIcon}
         ariaLabel={props.ariaLabel ?? tooltipContent ?? labelText}
         allowDisabledFocus={props.allowDisabledFocus ?? true}
+        aria-roledescription={props.ariaDescription}
```
Comment (JamesBurnside) [inline packages/react-components/src/components/ControlBarButton.tsx:147]: why `aria-roledescription` over `aria-description` ?

File: packages/react-components/src/components/MicrophoneButton.tsx [hunk 5]
```diff
@@ -166,6 +176,11 @@ export const MicrophoneButton = (props: MicrophoneButtonProps): JSX.Element => {
       menuProps={props.menuProps ?? generateDefaultDeviceMenuPropsTrampoline(props, strings)}
       menuIconProps={props.menuIconProps ?? !enableDeviceSelectionMenuTrampoline(props) ? { hidden: true } : undefined}
       split={props.split ?? enableDeviceSelectionMenuTrampoline(props)}
+      ariaDescription={
+        enableDeviceSelectionMenuTrampoline(props)
+          ? strings.microphoneButtonSplitRoleDescription
+          : strings.microphoneButtonDefaultRoleDescription
```
Comment (JamesBurnside) [inline packages/react-components/src/components/MicrophoneButton.tsx:182]:
```suggestion
         : undefined;
```

**Review by JamesBurnside — APPROVED** [review]

**Review by prprabhu-ms — APPROVED** [review]

**Pull request merged**